### PR TITLE
support applicaton snapshot input for sign off

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: internal/applicationsnapshot/input_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		t.Run(fmt.Sprintf("DetermineInputSpec=%d", i), func(t *testing.T) {
+
+			got, err := DetermineInputSpec(tc.filePath, tc.input, tc.imageRef)
+			if err != nil {
+				t.Fatalf("error: %v", err)
+			} else if !assert.ObjectsAreEqual(tc.want, got) {
+				t.Fatalf("got %v; want %v", got, tc.want)
+			} else {
+				t.Logf("Success!")
+			}
```
Comment (zregvart) [inline internal/applicationsnapshot/input_test.go:89]: This gives us nicer messages on test failures and is also a bit terser
```suggestion
			assert.NoError(t, err)
			assert.Equal(t, tc.want, got)
```

File: internal/image/build.go [hunk 2]
```diff
@@ -97,16 +97,20 @@ func (a *attestation) NewSignOffSource() (signOffSource, error) {
 // get the last commit used for the component build
 func (a *attestation) getBuildCommitSha() string {
 	if len(a.Predicate.Materials) == 1 {
-		return a.Predicate.Materials[0].Digest["sha"]
+		fmt.Printf("Using commit sha: %s", a.Predicate.Materials[0].Digest["sha1"])
```
Comment (zregvart) [inline internal/image/build.go]: This should use the log (import `log "github.com/sirupsen/logrus"`):
```suggestion
		log.Debug("Using commit sha: %s", a.Predicate.Materials[0].Digest["sha1"])
```

File: internal/image/build.go [hunk 3]
```diff
@@ -97,16 +97,20 @@ func (a *attestation) NewSignOffSource() (signOffSource, error) {
...
+		return a.Predicate.Materials[0].Digest["sha1"]
 	}
+	fmt.Println("did not find commit sha")
```
Comment (zregvart) [inline internal/image/build.go]:
```suggestion
	log.Debug("did not find commit sha")
```

File: internal/image/build.go [hunk 4]
```diff
@@ -97,16 +97,20 @@ func (a *attestation) NewSignOffSource() (signOffSource, error) {
...
 	return ""
 }
 
 // the git url used for the component build
 func (a *attestation) getBuildSCM() string {
 	if len(a.Predicate.Materials) == 1 {
+		fmt.Printf("Using uri: %s", a.Predicate.Materials[0].Uri)
```
Comment (zregvart) [inline internal/image/build.go]:
```suggestion
		log.Debug("Using uri: %s", a.Predicate.Materials[0].Uri)
```

File: internal/image/build.go [hunk 5]
```diff
@@ -97,16 +97,20 @@ func (a *attestation) NewSignOffSource() (signOffSource, error) {
...
 		return a.Predicate.Materials[0].Uri
 	}
+	fmt.Println("Did not find uri")
```
Comment (zregvart) [inline internal/image/build.go]:
```suggestion
	log.Debug("Did not find uri")
```

**Review by robnester-rh — COMMENTED** [review]
One comment regarding using `context` with git clone (something that @zregvart implemented as we rightly should consider context and time to clone, etc).

File: internal/image/build.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	"net/mail"
+	"regexp"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+)
+
+var gitClone = git.Clone
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply internal/image/build.go:31]: is `git.PlainClone` OK also?

**Review by robnester-rh — APPROVED** [review]

**Pull request merged**